src/main/ruby/truffleruby/core/io.rb

```diff
@@ -1648,6 +1650,18 @@ def printf(fmt, *args)
     write sprintf(fmt, *args)
   end
 
+  def timeout
+    @timeout || nil
+  end
+
+  def timeout=(new_timeout)
+    if Primitive.nil?(timeout) ^ Primitive.nil?(new_timeout)
```

Comment:
This seems hard to read and probably incorrect. I'd say we should nonblock = true if new_timeout is not nil.
The user might set/unset nonblock in between, we shouldn't rely on the old timeout value.

Reply:
True, updating.